File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums)-1
        if nums[r] > nums[l]:
            return nums[l]
        while r - l > 1:
            print(l,r)
            c = (r + l) // 2
            if nums[c] > nums[r]:
                l = c
            else:
                r = c
        return nums[r]


        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> hist;
        for (auto n : nums){
            hist[n]++;
        }
        priority_queue<pair<int, int>> shist;
        for (auto dataPair : hist){
            shist.push({dataPair.second, dataPair.first});
        }
        vector<int> res;
        for (int i=0; i<k; i++) {
            res.push_back(shist.top().second);
            shist.pop();
        }
        return res;
    }
};
